### PR TITLE
fix(#8617): Reduce enzyme usage in unit test by 25% - [2/3]

### LINKED ISSUE #8617: Reduce enzyme usage in unit test by 25% - [2 of 3]
### What is this about?

Convert the list of unit tests to use `react-testing-library` instead of `enzyme`. In total there are 53 files to convert. To make PRs easier to review and more manageable, this covers 17 test files.

### Scenario

_No response_

### Design

_No response_

### Technical Details

Convert these files

```
'app/components/UI/ActionView/index.test.tsx',
'app/components/UI/ActionModal/index.test.tsx',
'app/components/Views/confirmations/components/CustomNonce/index.test.tsx',
'app/components/Views/confirmations/SendFlow/ErrorMessage/index.test.tsx',
'app/components/Views/Settings/Contacts/AmbiguousAddressSheet/AmbiguousAddressSheet.test.tsx',
'app/components/Views/confirmations/components/Approval/ApprovalFlowLoader/index.test.tsx',
'app/components/UI/ImageIcon/index.test.tsx',
'app/components/UI/FadeOutOverlay/index.test.tsx',
'app/components/UI/AssetIcon/index.test.tsx',
'app/components/UI/TimeEstimateInfoModal/index.test.tsx',
'app/components/UI/Tabs/index.test.tsx',
'app/components/UI/ModalNavbarTitle/index.test.tsx',
'app/components/Views/WalletConnectSessions/index.test.tsx',
'app/component-library/components/Tags/Tag/Tag.test.tsx',
'app/components/UI/SwitchCustomNetwork/index.test.tsx',
'app/component-library/components/Avatars/Avatar/variants/AvatarAccount/AvatarAccount.test.tsx',
'app/components/Views/ImportPrivateKey/index.test.tsx',
```

### Threat Modeling Framework

_No response_

### Acceptance Criteria

Unit tests for the updated files should all pass. Snapshots should not change.

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by timi-codes**
## **Description**

Convert the list of unit tests to use react-testing-library instead of enzyme for the following files:

  ```
  'app/components/UI/ActionView/index.test.tsx',
  'app/components/UI/ActionModal/index.test.tsx',
  'app/components/Views/confirmations/components/CustomNonce/index.test.tsx',
  'app/components/Views/confirmations/SendFlow/ErrorMessage/index.test.tsx',
  'app/components/Views/Settings/Contacts/AmbiguousAddressSheet/AmbiguousAddressSheet.test.tsx',
  'app/components/Views/confirmations/components/Approval/ApprovalFlowLoader/index.test.tsx',
  'app/components/UI/ImageIcon/index.test.tsx',
  'app/components/UI/FadeOutOverlay/index.test.tsx',
  'app/components/UI/AssetIcon/index.test.tsx',
  'app/components/UI/TimeEstimateInfoModal/index.test.tsx',
  'app/components/UI/Tabs/index.test.tsx',
  'app/components/UI/ModalNavbarTitle/index.test.tsx',
  'app/components/Views/WalletConnectSessions/index.test.tsx',
  'app/component-library/components/Tags/Tag/Tag.test.tsx',
  'app/components/UI/SwitchCustomNetwork/index.test.tsx',
  'app/component-library/components/Avatars/Avatar/variants/AvatarAccount/AvatarAccount.test.tsx',
  'app/components/Views/ImportPrivateKey/index.test.tsx'
```

## **Related issues**

Fixes: #8617 

## **Manual testing steps**

Not Applicable

## **Screenshots/Recordings**

Not Applicable

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by timi-codes** [comment]
@Cal-L This is also ready for review

File: app/components/Views/Settings/Contacts/AmbiguousAddressSheet/AmbiguousAddressSheet.test.tsx [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+      navigate: jest.fn(),
+    }),
+  };
+});
+
 describe('AmbiguousAddressSheet', () => {
   it('should render correctly', () => {
-    const wrapper = shallow(<AmbiguousAddressSheet />);
-    expect(wrapper).toMatchSnapshot();
+    const { toJSON } = renderWithProvider(<AmbiguousAddressSheet />);
```
Comment (Cal-L) [inline app/components/Views/Settings/Contacts/AmbiguousAddressSheet/AmbiguousAddressSheet.test.tsx:31]: Replace with `renderScreen` and remove the `@react-navigation/native` mock. For the name, you can use `Routes.SHEET.AMBIGUOUS_ADDRESS`

File: app/components/Views/WalletConnectSessions/index.test.tsx [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+import { waitFor } from '@testing-library/react-native';
 import WalletConnectSessions from './';
+import AsyncStorage from '../../../store/async-storage-wrapper';
+import renderWithProvider from '../../../util/test/renderWithProvider';
 
 describe('WalletConnectSessions', () => {
-  it('should render correctly', () => {
-    const wrapper = shallow(
+  it('does not render when not ready', () => {
+    const { toJSON } = renderWithProvider(
```
Comment (Cal-L) [inline app/components/Views/WalletConnectSessions/index.test.tsx:9]: Replace with `renderScreen`

File: app/components/Views/WalletConnectSessions/index.test.tsx [hunk 3]
```diff
@@ -1,13 +1,55 @@
...
+      <WalletConnectSessions navigation={{ setOptions: () => null }} />,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('renders empty component with no active sessions', async () => {
+    jest.spyOn(AsyncStorage, 'getItem').mockResolvedValue(null);
+
+    const { getByTestId, toJSON } = renderWithProvider(
```
Comment (Cal-L) [inline app/components/Views/WalletConnectSessions/index.test.tsx:18]: Replace with `renderScreen`

File: app/components/Views/WalletConnectSessions/index.test.tsx [hunk 4]
```diff
@@ -1,13 +1,55 @@
...
+      <WalletConnectSessions navigation={{ setOptions: () => null }} />,
+    );
+
+    await waitFor(() => {
+      const emptyMessage = getByTestId('wallet-connect-sessions-screen');
```
Comment (Cal-L) [inline app/components/Views/WalletConnectSessions/index.test.tsx:23]:
```suggestion
      const emptyMessage = getByTestId(ExperimentalSelectorsIDs.CONTAINER);
```

You'll have to import ExperimentalSelectorsIDs

File: app/components/Views/WalletConnectSessions/index.test.tsx [hunk 5]
```diff
@@ -1,13 +1,55 @@
...
+        peerId: 'peer2',
+        peerMeta: { name: 'Session 2', url: 'https://example.org' },
+      },
+    ];
+
+    jest
+      .spyOn(AsyncStorage, 'getItem')
+      .mockResolvedValue(JSON.stringify(sessions));
+
+    const { getByTestId, toJSON } = renderWithProvider(
```
Comment (Cal-L) [inline app/components/Views/WalletConnectSessions/index.test.tsx:45]: Replace with `renderScreen`

File: app/components/Views/WalletConnectSessions/index.test.tsx [hunk 6]
```diff
@@ -1,13 +1,55 @@
...
       <WalletConnectSessions navigation={{ setOptions: () => null }} />,
     );
 
-    expect(wrapper).toMatchSnapshot();
+    await waitFor(() => {
+      const viewID = getByTestId('wallet-connect-sessions-screen');
```
Comment (Cal-L) [inline app/components/Views/WalletConnectSessions/index.test.tsx:50]:
```suggestion
      const viewID = getByTestId(ExperimentalSelectorsIDs.CONTAINER);
```

**Review by Cal-L — COMMENTED** [review]
@timi-codes  Left some comments. Let's also resolve the conflicts

**Comment by timi-codes** [comment]
@Cal-L Thanks for the feedback. I think /@kylanhurt already worked on the same files. Reason for the conflict. However, I already pulled his changes into this branch. Now we have just a few changes

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**